Add discord bot which integrates with the existing NER model

pncnmnp opened the pull request:
The bot contains logging identification and basic name tagging functionality

pncnmnp committed:
Add discord bot which integrates with the existing NER model
The bot contains logging identification and basic name tagging functionality

CallmeMehdi commented:
Everything looks good @pncnmnp it works perfectly. But can you please change the branch to dev? We need to merge discord into dev.

pncnmnp commented:
Changed base to dev

pncnmnp commented on src/discord_bot.py:
```diff
@@ -0,0 +1,66 @@
+        content = message.content.replace("$tag", "")
+        prediction = guild_obj.bento_service.predict([{"sentence": content}])
+
+        tokens = prediction[0].split(",")
```
They are not integrated!

pncnmnp commented:
**This pull request should not be merged. The integration part with NER is incomplete.**
I have not deployed the tokens in line 50 of discord_bot.py.

pncnmnp committed:
Combine model predictions with tagging process
All nickname components will be considered separately
For example: "Parth Parikh" will be treated as ["Parth", "Parikh"]
to reduce false negatives while tagging

pncnmnp reviewed (commented):
I have fixed the changes. @CallmeMehdi can you review them.

CallmeMehdi commented on src/discord_bot.py:
```diff
@@ -0,0 +1,70 @@
+# Store in a .env file with the following structure
+"""
+# .env
+DISCORD_TOKEN={your-discord-token}
```
This will be then changed in the deployment right?

pncnmnp replied:
Yes we need to create a `.env` file and add the tokens there without making a commit of `.env`.

CallmeMehdi reviewed (approved):
Everything seems great!

CallmeMehdi commented:
We don't need to delete the discord branch as we will work on it in the future.
